boxoffice/models/line_item.py

```diff
@@ -235,36 +237,61 @@ def counts_per_date_per_item(item_collection, user_tz):
     return date_item_counts
 
 
-def sales_by_date(dates, user_tz, item_ids):
+def sales_by_date(sales_datetime, item_ids):
     """
-    Returns the net sales of line items sold on a date.
-    Accepts a list of dates.
-    ['2016-01-01', '2016-01-02'] => {'2016-01-01': }
+    Takes a timezone aware datetime object, item_ids and returns the sales amount accrued
+    during the day for the given datetime.
     """
     if not item_ids:
         return None
 
-    date_sales = {}
+    utc_start_at = reset_time_utc(sales_datetime, datetime.datetime.min.time())
+    utc_end_at = reset_time_utc(sales_datetime, datetime.datetime.max.time())
+    sales_on_date = db.session.query('sum').from_statement('''SELECT SUM(final_amount) FROM line_item
+        WHERE status=:status AND ordered_at >= :start_at AND ordered_at <= :end_at
+        AND line_item.item_id IN :item_ids
+        ''').params(status=LINE_ITEM_STATUS.CONFIRMED, start_at=utc_start_at, end_at=utc_end_at, item_ids=tuple(item_ids)).first()
+    return sales_on_date[0] if sales_on_date[0] else Decimal(0)
 
-    for sales_date in dates:
-        sales_on_date = db.session.query('sum').from_statement('''SELECT SUM(final_amount) FROM line_item
-            WHERE status=:status AND DATE_TRUNC('day', line_item.ordered_at AT TIME ZONE 'UTC' AT TIME ZONE :timezone)::date = :sales_date
-            AND line_item.item_id IN :item_ids
-            ''').params(timezone=user_tz, status=LINE_ITEM_STATUS.CONFIRMED, sales_date=sales_date, item_ids=tuple(item_ids)).first()
-        date_sales[sales_date] = sales_on_date[0] if sales_on_date[0] else Decimal(0)
-    return date_sales
 
-
-def sales_delta(user_tz, item_ids):
+def calculate_weekly_sales(item_collection_ids, user_tz, year):
     """
-    Calculates the percentage difference in sales between today and yesterday
+    Calculates sales per week for items in the given set of item_collection_ids in a given year,
+    in the user's timezone.
     """
-    today = datetime.datetime.now().strftime("%Y-%m-%d")
-    yesterday = (datetime.datetime.now() - datetime.timedelta(days=1)).strftime("%Y-%m-%d")
-    sales = sales_by_date([today, yesterday], user_tz, item_ids)
-    if not sales or not sales[yesterday]:
+    ordered_week_sales = OrderedDict()
+    local_start_at = reset_time_local(Week.monday(Week(year, 1)), user_tz)
```

Comment:
`Week.monday` returns a `date`. This `date` is converted to a `datetime` object in the user's timezone, which is then casted to `utc` (in the next line) for it to be used in the query.